File: selfstudy/js/cities_reverse.js
/*
task 78:
Cities - Reverse
You need to reverse the order of city populations in thousands
*/

function reverse_order_cities(populations) {
  return [...populations].reverse();
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: reverse_order_cities([10, 20, 30]) },
  { input: [50, 60, 70], expected: reverse_order_cities([50, 60, 70]) },
  { input: [1, 2, 3], expected: reverse_order_cities([1, 2, 3]) },
  { input: [100, 200], expected: reverse_order_cities([100, 200]) },
  { input: [], expected: reverse_order_cities([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('reverse_order_cities', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(reverse_order_cities(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, reverse_order_cities(t.input) === t.expected || JSON.stringify(reverse_order_cities(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// finally works

File: selfstudy/golang/cities_reverse.go
/*
task 278:
Cities - Reverse
You need to reverse the order of city populations in thousands
*/

package main

import (
	"fmt"
	"reflect"
)

func Reverse_orderCities(populations []int) []int {
	res := make([]int, len(populations))
	for i, v := range populations {
		res[len(populations)-1-i] = v
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Reverse_orderCities([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Reverse_orderCities([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Reverse_orderCities([]int{1, 2, 3})},
		{[]int{100, 200}, Reverse_orderCities([]int{100, 200})},
		{[]int{}, Reverse_orderCities([]int{})},
	}
	
	for idx, t := range tests {
		res := Reverse_orderCities(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// finally works
